### PR TITLE
Add `hello world` flow to prefect module for very fast onboarding

### PR DESCRIPTION
**Pull request by madkinsz**
## Summary

This adds a very basic hello world flow to the Prefect module itself. I use a hello world flow _all the time_ to test things and I think it's super reasonable to provide one immediately to the user. I think this is the _only_ example we should inline in this way. This allows onboarding to look like:

```
❯ prefect register --project default --module prefect.hello_world
Collecting flows...
Processing 'prefect.hello_world':
  Building `Module` storage...
  Registering 'hello-world'... Done
  └── ID: bfc72a71-8732-4c74-a119-e4a3b1309f00
  └── Version: 5
======================== 1 registered ========================

❯ prefect run --module prefect.hello_world -p name \"Marvin\" --watch
Looking up flow... Done
Creating run for flow 'hello-world'... Done
└── Name: mauve-piculet
└── UUID: 48721132-d050-48d0-8188-78b31b61341c
└── Labels: []
└── Parameters: {'name': 'Marvin'}
└── Context: {}
└── URL: https://cloud.prefect.io/michael-prefect-personal/flow-run/48721132-d050-48d0-8188-78b31b61341c
Watching flow run execution...
└── 11:29:14 | INFO    | Entered state <Pending>: Flow run created
└── 11:29:14 | INFO    | Entered state <Scheduled>: Flow run scheduled.
└── 11:29:19 | INFO    | Entered state <Submitted>: Submitted for execution
└── 11:29:19 | INFO    | Submitted for execution: PID: 75974
└── 11:29:20 | INFO    | Entered state <Running>: Running flow.
└── 11:29:19 | INFO    | Beginning Flow run for 'hello-world'
└── 11:29:21 | INFO    | Task 'name': Starting task run...
└── 11:29:21 | INFO    | Task 'name': Finished task run for task with final state: 'Success'
└── 11:29:22 | INFO    | Task 'say_hello': Starting task run...
└── 11:29:22 | INFO    | Hello Marvin
└── 11:29:22 | INFO    | Task 'say_hello': Finished task run for task with final state: 'Success'
└── 11:29:22 | INFO    | Flow run SUCCESS: all reference tasks succeeded
└── 11:29:23 | INFO    | Entered state <Success>: All reference tasks succeeded.
Flow run succeeded!
```

Note this example requires #4463

## Changes

- Adds `prefect.hello_world` module with a single parameter flow

## Importance

- Improves onboarding with an immediate flow to play with
- I personally use this flow constantly to test features

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/hello_world.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import prefect
+
+
+@prefect.task(log_stdout=True)
+def say_hello(name: str) -> None:
+    print(f"Hello {name}")
+
+
+with prefect.Flow("hello-world") as flow:
```
Comment (madkinsz) [inline src/prefect/hello_world.py:17]: Should the variable be called `hello_flow` ?

File: src/prefect/hello_world.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"""
+A very basic 'hello-world' flow for getting started quickly
+
+Examples:
+    > prefect register --project default -m prefect.hello_world
+    > prefect run -m prefect.hello_world --no-agent
```
Comment (madkinsz) [inline src/prefect/hello_world.py:6]: I may exclude this example until this feature is ready

Comment (zangell44) [reply src/prefect/hello_world.py:6]: +1 would vote to exclude until this is ready;

**Review by zangell44 — CHANGES_REQUESTED** [review]
one nit, looks good to me

**Review by zangell44 — APPROVED** [review]

**Pull request merged**